packages/block-library/src/comments/index.php

```diff
@@ -0,0 +1,238 @@
+			'wp-block-buttons',
+			'wp-block-button',
+		);
+		foreach ( $handles as $handle ) {
+			wp_enqueue_block_style( $block_name, array( 'handle' => $handle ) );
+		}
+		$are_styles_enqueued = true;
+	}
+}
+
+/**
+ * Renders the legacy `core/post-comments` block on the server.
+ * It triggers a developer warning and then calls the renamed
+ * block's `render_callback` function output.
+ *
+ * @param array    $attributes Block attributes.
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ *
+ * @return string Returns the output of the block.
+ */
+function render_legacy_post_comments_block( $attributes, $content, $block ) {
+	trigger_error(
+		/* translators: %1$s: Block type */
+		sprintf( __( 'Block %1$s has been renamed to Comments Query Loop. %1$s will be supported until WordPress version X.X.', 'gutenberg' ), $block->name ),
+		headers_sent() || WP_DEBUG ? E_USER_WARNING : E_USER_NOTICE
+	);
+	return render_block_core_comments( $attributes, $content, $block );
+}
+
+/**
+ * Ensures backwards compatibility for any users running the Gutenberg plugin
+ * who have used Post Comments before it was merged into Comments Query Loop.
+ *
+ * The same approach was followed when core/query-loop was renamed to
+ * core/post-template.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/41807
+ * @see https://github.com/WordPress/gutenberg/pull/32514
+ */
+function register_legacy_post_comments_block() {
+	$registry = WP_Block_Type_Registry::get_instance();
+
+	/*
+	 * Remove the old `post-comments` block if it was already registered, as it
+	 * is about to be replaced by the type defined below.
+	 */
+	if ( $registry->is_registered( 'core/post-comments' ) ) {
+		unregister_block_type( 'core/post-comments' );
+	}
+
+	// Recreate the legacy block metadata.
+	$metadata = array(
+		'name'              => 'core/post-comments',
+		'category'          => 'theme',
+		'attributes'        => array(
+			'textAlign' => array(
+				'type' => 'string',
+			),
+		),
+		'uses_context'      => array(
+			'postId',
+			'postType',
+		),
+		'supports'          => array(
+			'html'       => false,
+			'align'      => array( 'wide', 'full' ),
+			'typography' => array(
+				'fontSize'                      => true,
+				'lineHeight'                    => true,
+				'__experimentalFontStyle'       => true,
+				'__experimentalFontWeight'      => true,
+				'__experimentalLetterSpacing'   => true,
+				'__experimentalTextTransform'   => true,
+				'__experimentalDefaultControls' => array(
+					'fontSize' => true,
+				),
+			),
+			'color'      => array(
+				'gradients'                     => true,
+				'link'                          => true,
+				'__experimentalDefaultControls' => array(
+					'background' => true,
+					'text'       => true,
+				),
+			),
+			'inserter'   => false,
+		),
+		'style'             => array(
+			'wp-block-post-comments',
+			'wp-block-buttons',
+			'wp-block-button',
```

Comment:
Why do we need these styles here? 🤔

Reply:
They were already defined in `core/post-comments`'s metadata, so I kept them. In fact, at least the `wp-block-button` class appears in the rendered output. Would it be better to remove them?

Reply:
Ah yeah, I wasn't really familiar with this technique.

Well in that case, we should keep the `wp-block-button` line; I'm not sure we need `wp-block-buttons` though -- I haven't seen any reference to it in the PHP-generated markup from the block.